backend/Cargo.toml

```diff
@@ -31,3 +31,4 @@ reqwest = "0.11.14"
 alcoholic_jwt = "4091.0.0"
 thiserror = "1.0.*"
 chrono = { version = "0.4.23", features = ["serde"] }
+names = { version = "0.14.0", default-features = false }
```

Comment:
Some random things I have noticed:

- I think when you create  a challenge we should bring you to the "my challenges" tab, it currently just disappears and I was confused for a second before I noticed I was still on the "watch" tab
- I needed to change the `next.config.js` to `destination: 'http://127.0.0.1:8080/api/:path*',`
- some of the error handling could implement a `From` then we don't need to do `map_err` all the time